File: Backend/EventManager/EventManager.Domain/Models/JobOffer.cs
using EventManager.Domain.Enums;

namespace EventManager.Domain.Models
{
    public class JobOffer
    {
        public int Id { get; set; }
        public PositionType PositionType { get; set; }
        public Event Event { get; set; }
        public double Salary { get; set; }
        public double Hour { get; set; }

        public void Update(JobOffer offer)
        {
            PositionType = offer.PositionType;
            Event = offer.Event;
            Salary = offer.Salary;
            Hour = offer.Hour;
        }
    }
}

File: Frontend/event-manager/src/app/shared/models/job-offer.ts
import { PositionType } from './../enums/position-type';
export class JobOffer {
  id: number;
  positionType: PositionType;
  event: Event;
  salary: number;
  hour: number;

  constructor(id: number, positionType: PositionType, event: Event, salary: number, hour: number){
    this.id = id;
    this.positionType = positionType;
    this.event = event;
    this.salary = salary;
    this.hour = hour;
  }
}
